contracts/TwoUserMultisig.sol

```diff
@@ -121,27 +113,109 @@ contract TwoUserMultisig is IAccount, IERC1271 {
         payable
     {
         _validateTransaction(bytes32(0), _transaction);
-
         _executeTransaction(_transaction);
     }
 
-    function isValidSignature(bytes32 _hash, bytes calldata _signature)
+    function isValidSignature(bytes32 _hash, bytes memory _signature)
         public
         view
         override
-        returns (bytes4)
+        returns (bytes4 magic)
     {
-        // The signature is the concatenation of the ECDSA signatures of the owners
-        // Each ECDSA signature is 65 bytes long. That means that the combined signature is 130 bytes long.
-        require(_signature.length == 130, "Signature length is incorrect");
+        magic = EIP1271_SUCCESS_RETURN_VALUE;
+
+        if (_signature.length != 130) {
+            // Signature is invalid anyway, but we need to proceed with the signature verification as usual
+            // in order for the fee estimation to work correctly
+            _signature = new bytes(130);
+            
+            // Making sure that the signatures look like a valid ECDSA signature and are not rejected rightaway
+            // while skipping the main verification process.
+            _signature[64] = bytes1(uint8(27));
+            _signature[129] = bytes1(uint8(27));
```

Comment:
this signature substitution logic happens when called from `validateTransaction`, but also during regular EIP1271  read calls.

is this intended?

Reply:
@delaaxe it was just more convenient in this particular code, but no, it is not required. The substitution could be moved to other places